### ISSUE TITLE
Why tuning hyper-parameter changes the mean_module?

### ISSUE THREAD

**Issue by guol1nag** [cornellius-gp/gpytorch#1282]
Hi,

Why tuning hyper-parameters by maximising the Exact Log Marginal Likelihood also changes the gpytorch.means.ConstantMean? I don't think Exact Log Marginal Likelihood is a function of the mean

Thanks,

**Comment by KeAWang** [comment]
As I said in https://github.com/cornellius-gp/gpytorch/issues/1279#issuecomment-689024496, if you have a non-zero prior mean, then the log marginal likelihood will involve the mean. If you have a mean m(x), the log marginal likelihood will be
log p(f(x) | X) = -(1/2) (f - m(x))^T K^{-1} (f - m(x)) - (1/2) log |K| - (n/2) log (2 \pi)

In this case, `ConstantMean` is a constant (but learnable) mean. If you want to set a fixed mean that doesn't change, you can just detach it from the graph.

```
prior_mean = ConstantMean()
prior_mean.initialize(constant=your_constant_tensor)
prior_mean.requires_grad = False
```

**Comment by guol1nag** [comment]
Thanks! That is very clear!
